dotnet/src/Agents/Core/Chat/ExpressionTerminationStrategy.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Agents.Chat;
+
+/// <summary>
+/// Signals termination when the most recent message matches against the defined regular expressions
+/// for the specified agent (if provided).
+/// </summary>
+public sealed class ExpressionTerminationStrategy : AgentBoundTerminationStrategy
```

Comment:
How about naming this `RegexTerminationStrategy` for improved clarity?

Reply:
I would add the `Agents` property directly here

Reply:
Yes, on renaming.

In response to `Agents` property, I presume you are referring to the `AgentBoundTerminationStrategy` base class.

The reason `AgentBoundTerminationStrategy` exists is that it will be a common case for a `TerminationStrategy` to be scoped to an agent (encapsulation opportunity).  With that said, looking at this with fresh eyes and the entire contract can be pushed down into `TerminationStrategy`.  This is the opposite of your suggestion, but feels like the right direction.

(will iterate in that direction along with the default strategies)

Reply:
Removed `AgentBoundTerminationStrategy`